### PR TITLE
migration: Offer option to delete VM storage after migration from source VM

### LINKED ISSUE #165: Delete storage from origin host after migration
Created from @garrett comment in #15 

> 117332031-a535e700-ae97-11eb-8538-8475268e6fc4
> 
> The checkbox is "progressive disclosure" and only appears when the radio above is selected. But, again: This shouldn't land in this PR. I'm just finding a place for it in the design so we don't have to "shoehorn" it in later. (That is, so we don't have to find a place to shove it in without having consideration in the first place.)

### PR DESCRIPTION
**Pull request by skobyda**
Only shows if user doesn't select temporation nor shared storage.

Fixes #165

Screenshot from 2021-07-05 12-57-34

### CODE REVIEW

**Comment by KKoukiou** [comment]
Test failure.

**Comment by skobyda** [comment]
Test failure fixed

File: src/components/vm/vmMigrateDialog.scss [hunk 1]
```diff
@@ -6,3 +6,7 @@
 .popover-headline {
     margin-bottom: var(--pf-global--spacer--xs);
 }
+
+#delete-storage {
```
Comment (KKoukiou) [inline src/components/vm/vmMigrateDialog.scss:10]: I am not fond of custom CSS for nesting form fields. You can have some kind of nesting with the following code:
``` diff
diff --git src/components/vm/vmMigrateDialog.jsx src/components/vm/vmMigrateDialog.jsx
index 2a41678c..d96661b4 100644
--- src/components/vm/vmMigrateDialog.jsx
+++ src/components/vm/vmMigrateDialog.jsx
@@ -110,7 +110,8 @@ const StorageRow = ({ storage, setStorage, temporary, deleteStorage, setDeleteSt
                                <OutlinedQuestionCircleIcon />
                            </button>
                        </Popover>
-                   }>
+                   }
+                   isStack>
             <Radio id="shared"
                    name="storage"
                    label={_("Storage is at a shared location")}
@@ -120,13 +121,15 @@ const StorageRow = ({ storage, setStorage, temporary, deleteStorage, setDeleteSt
                    name="storage"
                    label={_("Copy storage")}
                    isChecked={storage === "copy"}
-                   onChange={() => setStorage("copy")} />
-            {storage === "copy" && !temporary &&
-                <Checkbox id="delete-storage"
-                          isChecked={deleteStorage}
-                          label={_("Delete storage from origin host after migration")}
-                          onChange={value => setDeleteStorage(value)} />
-            }
+                   onChange={() => setStorage("copy")}
+                   body={
+                       (storage === "copy" && !temporary)
+                           ? <Checkbox id="delete-storage"
+                                       isChecked={deleteStorage}
+                                       label={_("Delete storage from origin host after migration")}
+                                       onChange={value => setDeleteStorage(value)} />
+                           : null
+                   } />
         </FormGroup>
     );
 };
diff --git src/components/vm/vmMigrateDialog.scss src/components/vm/vmMigrateDialog.scss
index 9c8a1dc7..28927a75 100644
--- src/components/vm/vmMigrateDialog.scss
+++ src/components/vm/vmMigrateDialog.scss
@@ -6,7 +6,3 @@
 .popover-headline {
     margin-bottom: var(--pf-global--spacer--xs);
 }
-
-#delete-storage {
-    margin-left: var(--pf-global--spacer--xl);
-}

```

**Review by KKoukiou — CHANGES_REQUESTED** [review]
The code changes are fine. I would like that we find a better way for nesting the options however. Sent a patch to try out.

File: src/components/vm/vmMigrateDialog.scss (hunk 1 above)
Comment (skobyda) [reply src/components/vm/vmMigrateDialog.scss]: Thanks, works like charm

**Comment by skobyda** [comment]
Screenshot from 2021-07-05 12-57-34

**Review by KKoukiou — APPROVED** [review]

**Comment by garrett** [comment]
Should this be a release note? (If so, we should replace the top image with the bottom one and add a release note tag.)

**Comment by skobyda** [comment]
> Should this be a release note? (If so, we should replace the top image with the bottom one and add a release note tag.)

The migration itself was still not put into release notes yet, as there were some bugs. I sent fixes, so hopefully we can introduce migration in release notes this release. 
So maybe we can advertise this functionality together with a post about the migration itself?
Adding tas so we don't forget about that

**Comment by KKoukiou** [comment]
I think we can close this PR @skobyda, see https://github.com/cockpit-project/cockpit-machines/issues/247 

Sorry that  you spent time on this :(

**Comment by skobyda** [comment]
It seems so, since we are getting rid of Copy storage option altogether, this PR is no longer needed.

**Pull request closed without merging**